### ISSUE TITLE
Description of impurities in the wire

### ISSUE THREAD

**Issue by terracoda** [phetsims/resistance-in-a-wire#140]
In a recent interview, a blind user did not understand the part of the Wire's description that describes impurities.

> Currently, wire is {{very short}}, {{thick}}, and {{has a large amount of impurities}}.

Wondering if we should add the follow static string to the end of the wire's description, "represented by black dots."

**Examples:**
> Currently, wire is {{very short}}, {{thick}}, and {{has a large amount of impurities}} represented by black dots.

> Currently, wire is {{very long}}, {{thin}}, and {{has a very small amount of impurities}} represented by black dots.

> Currently, wire is {{long}}, {{extremely thin}}, and {{has a tiny amount of impurities}} represented by black dots.

If this change is added, it would also affect Ohm's Law.

**Comment by terracoda** [comment]
@emily-phet and @arouinfar, I think in my first draft of descriptions I just described the amount of dots and then re-worded based on feedback from @arouinfar.

Maybe we need both words to connect the concept of "impurities" to their actual visual representation of  "dots" in the wire?

**Comment by arouinfar** [comment]
> Maybe we need both words to connect the concept of "impurities" to their actual visual 
representation of "dots" in the wire?

@terracoda I really like this proposal! I think the addition of "represented by black dots" will make the impurities a bit more concrete for students, hopefully making the concept of resistivity a bit clearer. Let's proceed with your proposed changes to the descriptions.

**Comment by emily-phet** [comment]
@terracoda 
I'm concerned this may be too verbose, and too focused on the visual representation. Regardless of how the dots are represented, more or less impurities is the important idea.

Could you be more specific about what the participant specifically said/did that you interpreted as not understanding? Did they hear those and do something like...user says "i don't really get what that means" and move on. Or did they listen multiple times, try to make sense of it, and weren't able to without help...etc. No need to write too much here if it's easier just to describe it in realtime - we can talk about it sometime today. 

Also, what was this users background? I can totally understand if this was (for example) an older participant, long removed from a science class, who does not have work or hobbies with a science-y bend not grasping "impurities" in this context not for lack of understanding "represented by black dots" but for other reasons. But perhaps this was a scientist(!) or college student we might reasonably expect to have an appropriate background to understand this - in which case, it should definitely be more clear.

**Comment by terracoda** [comment]
@emily-phet, great questions.

In terms of verbosity, I agree that we should always be careful about verbosity. In this case, however, the issue is regarding a dynamic description, not an alert, so I think a little extra detail if needed is totally fine in a dynamic description of a changing sim object. It is in the dynamic description that a student would get details.

The participant is an undergrad with a science background. My recollection is that they read the wire description a couple of times and said something like, "I do not understand this bit about impurities."

I think - but would have to review to be sure - what was happening is that they were not connecting a visual in their mind in their own mental model of what "impurities" actually are. Again, I would have to review the recording to be sure.

**Comment by emily-phet** [comment]
@terracoda 
I'm going to make this suggestion below - but I completely get that during an interview you're getting lots of information (subtle and not-so-subtle) that informs your suggestion. If you feel strongly that adding "represented by black dots" would address the specific type of problem you were observing, please continue to insist on it! Ok - on to my suggestion.

Thanks for the info! Could you consider a way to rephrase the dynamic description so that the part indicating the impurities has "in the wire" right after? As in "There are a {{large amount of impurities}} in the wire". It feels a little more targeted to say "there are a large amount of impurities **in the wire**" in comparison to "the wire has a large amount of impurities". Does that make sense to you? Do you think that might have helped the student you interviewed? 

My hesitancy is that adding "represented by black dots" doesn't really get at the learning goal of being able to track that the amount of impurities changes as resistivity changes. This addition feels more like creating a helpful mapping between the visual and non-visual, which feels like an indicator that something bigger is off with the overall string.

**Comment by terracoda** [comment]
@emily-phet, I think that is a great suggestion! I'm going to iterate a little in the design document.

**Comment by terracoda** [comment]
@arouinfar, do you have a preference for any of the following examples? I've twekaed order and words in each example. Keep in mind, an h3-heading, "The Wire", starts of the description of the wire object. 

**_Example 1:_**
- Currently, wire is {{very short}}, {{thick}}, and in wire, there is {{a large amount of impurities}}.
- Currently, wire is {{very short}}, {{thick}}, and in wire, there is {{a small amount of impurities}}.
- Currently, wire is {{long}}, {{extremely thin}}, and in wire, there is {{a tiny amount of impurities}}.

**_Example 2:_**
- Currently, wire is {{very short}}, {{thick}}, and there is {{a large amount of impurities}} in wire.
- Currently, wire is {{very short}}, {{thick}}, and there is {{a small amount of impurities}} in wire.
- Currently, wire is {{long}}, {{extremely thin}}, and there is {{a tiny amount of impurities}} in wire.

**_Example 3: Two sentences_**
- Currently, wire is {{very short}} and {{thick}}. In wire, there is {{a large amount of impurities}}.
- Currently, wire is {{very short}} and {{thick}}. In wire, there is {{a small amount of impurities}}.
- Currently, wire is {{long}} and {{extremely thin}}. In wire, there is {{a tiny amount of impurities}}.

**_Example 4: Two sentences_**
- Currently, wire is {{very short}} and {{thick}}. There is {{a large amount of impurities}} in wire.
- Currently, wire is {{very short}} and {{thick}}. There is {{a small amount of impurities}} in wire.
- Currently, wire is {{long}} and {{extremely thin}}. There is {{a tiny amount of impurities}} in wire.

**_Example 5: Using "material of wire"_**
- Currently, wire is {{very short}}, {{thick}}, and material of wire has {{a large amount of impurities}}.
- Currently, wire is {{very short}}, {{thick}}, and material of wire has {{a small amount of impurities}}.
- Currently, wire is {{long}}, {{extremely thin}}, and material of wire has {{a tiny amount of impurities}}.

**_Example 6: Using "material of wire contains"_**
- Currently, wire is {{very short}}, {{thick}}, and material of wire contains {{a large amount of impurities}}.
- Currently, wire is {{very short}}, {{thick}}, and material of wire contains {{a small amount of impurities}}.
- Currently, wire is {{long}}, {{extremely thin}}, and material of wire contains {{a tiny amount of impurities}}.

**In comparison to the explicit description of "black dots":**
- Currently, wire is {{very short}}, {{thick}}, and {{has a large amount of impurities}} represented by black dots.
- Currently, wire is {{very long}}, {{thin}}, and {{has a very small amount of impurities}} represented by black dots.
- Currently, wire is {{long}}, {{extremely thin}}, and {{has a tiny amount of impurities}} represented by black dots.

**Comment by terracoda** [comment]
@arouinfar, I like @emily-phet suggestion because, it might be the case that in collaboration two students can make the connection between the visual black dots and what they represent through discussion rather than the PDOM providing an explicit description of the dots.

**Comment by terracoda** [comment]
@arouinfar, I'll also put this to the group in the A11y Design meeting today.

**Comment by emily-phet** [comment]
I like version 2 and 4.

**Comment by arouinfar** [comment]
> @arouinfar, I like @emily-phet suggestion because, it might be the case that in collaboration two students can make the connection between the visual black dots and what they represent through discussion rather than the PDOM providing an explicit description of the dots.

I agree. I think "in the wire" makes the description more concrete (which is why I liked your original suggestion). However, the fact that the impurities are represented by black dots isn't particularly important.

I prefer option (2), but if we decide to go for two sentences, then (4) would be my choice.

**Comment by terracoda** [comment]
This all sounds good.
I checked Ohm's Law, and realized we are covered there as there is a bit more detail:

> In circuit, 
-batteries supply {{4.5 volts}} 
-resistor shows {{a medium amount of impurities}}
-{{small arrows}} indicate a current flowing clockwise at {{5.0 millamps}}

**Comment by terracoda** [comment]
Ok, sounds good. Let's go with Example 2.
So to clarify, this a description change for the wire's dynamic description in the Parallel DOM. It does not affect alerts.

**Example 2:**
    Currently, wire is {{very short}}, {{thick}}, and there is {{a large amount of impurities}} in wire.
    Currently, wire is {{very short}}, {{thick}}, and there is {{a small amount of impurities}} in wire.
    Currently, wire is {{long}}, {{extremely thin}}, and there is {{a tiny amount of impurities}} in wire.

I'll update the design document with this change to make sure there are no unforeseen changes to the string parameters.

@jessegreenberg, please implement the change or assign to the developer who is working on this sim, thanks!

**Comment by terracoda** [comment]
@jessegreenberg, I have updated the design document for RIAW. This string change is ready for implementation. The parameters have not changed. Not that the indefinite article, "a", in the amount of impurities could be part of the string rather than the parameter. I'm not sure how you have set that up in the code files. I cleaned that up in the design document and removed "a" from the parameters.

**Comment by jessegreenberg** [comment]
@zepumph this is ready, would you mind working on this and merging into  the release branch?

**Comment by terracoda** [comment]
@zepumph, just an afterthought regarding where the indefinite article, "a", should go. For internationalization, it might actually be better placed within the parameter? I'm not sure about this.

**Comment by jessegreenberg** [comment]
@zepumph and I talked, I am going to work on this. There are issues from this RC that are caused by AT workarounds from issues I have worked on, and since the version is nearly ready it will add unnecessary overhead to have another developer pick up. Reassigning to me.

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/resistance-in-a-wire/issues/140#issuecomment-379235336

Thanks @terracoda, we will take a look.

**Comment by jessegreenberg** [comment]
This will need to be merged into release branch for https://github.com/phetsims/QA/issues/108

**Comment by zepumph** [comment]
@terracoda I almost missed your comments! Sorry about that.

**Comment by zepumph** [comment]
@lmulhall-phet can this be closed? Was it tested as part of phetsims/QA#118?
